app/dashboard/models.py

```diff
@@ -764,3 +764,21 @@ def normalize_tip_usernames(sender, instance, **kwargs):
 
 
 m2m_changed.connect(m2m_changed_interested, sender=Bounty.interested.through)
+
+
+class CoinRedemption(SuperModel):
+  shortcode = models.CharField(max_length=255, default='')
+  url = models.URLField(null=True)
+  network = models.CharField(max_length=255, default='')
+  token_name = models.CharField(max_length=255)
+  contract_address = models.CharField(max_length=255)
+  amount = models.IntegerField(default=1)
+  expires_date = models.DateTimeField()
+
+
+class CoinRedemptionRequest(SuperModel):
+  coin_redemption = models.OneToOneField(CoinRedemption, blank=False, on_delete=models.CASCADE)
+  ip = models.CharField(max_length=50)
```

Comment:
Just a note, there is an IP Address model field type. https://docs.djangoproject.com/en/2.0/ref/models/fields/#genericipaddressfield